engine/access/rpc/backend/backend.go

```diff
@@ -226,12 +230,80 @@ func (b *Backend) GetNetworkParameters(_ context.Context) access.NetworkParamete
 
 // GetLatestProtocolStateSnapshot returns the latest finalized snapshot
 func (b *Backend) GetLatestProtocolStateSnapshot(_ context.Context) ([]byte, error) {
-	data, err := convert.SnapshotToBytes(b.state.Final())
+	snapshot := b.state.Final()
+
+	validSnapshot, err := b.getValidSnapshot(snapshot, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	return convert.SnapshotToBytes(validSnapshot)
+}
+
+// getValidSnapshot will return a valid snapshot that has a sealing segment which
+// 1. does not contain any blocks that span an epoch transition
+// 2. does not contain any blocks that span an epoch phase transition
+// If a snapshot does contain an invalid sealing segment query the state
+// by height of each block in the segment and return a snapshot at the point
+// where the transition happens.
+func (b *Backend) getValidSnapshot(snapshot protocol.Snapshot, blocksVisited int) (protocol.Snapshot, error) {
+	if blocksVisited > snapshotHistoryLimit {
+		return nil, fmt.Errorf("failed to get valid snapshot reached look back limit ")
+	}
+
+	segment, err := snapshot.SealingSegment()
+	if err != nil {
+		return nil, err
```

Comment:
A good practice is always wrap the error with some messages, so that when the error occurs, we could see the call stacks along with some useful parameters.

Reply:
https://github.com/onflow/flow-go/pull/1940/commits/a24a5f1342adf1523fafcdbf08216a1465ec00b0